### PR TITLE
Implemeneted reciprocal operator

### PR DESCRIPTION
**Pull request by apaleyes**
As suggested in #3201

Besides, a few minor changes:
- Improved documentation for negative operator to be in line with numpy
- Adjusted indentation in one of the unit tests for operators

### CODE REVIEW

File: src/operator/tensor/elemwise_unary_op.cc [hunk 1]
```diff
@@ -171,10 +171,24 @@ NNVM_REGISTER_OP(_backward_cast)
...
+
+Calculates 1/x.
+
+Example::
+
+    reciprocal([-2, 1, 3, 1.6, 0.2]) = [-0.5, 1.0, 0.33333334, 0.625, 5.0]
+
+)code" ADD_FILELINE)
+.set_attr<FCompute>("FCompute<cpu>", UnaryCompute<cpu, mshadow_op::reciprocal>)
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseNone{"reciprocal"});
```
Comment (piiswrong) [inline src/operator/tensor/elemwise_unary_op.cc]: gradient is wrong.

File: tests/python/unittest/test_operator.py [hunk 2]
```diff
@@ -3106,17 +3106,32 @@ def test_ctc_loss():
...
+
+def test_reciprocal_op():
+    data_tmp = np.random.rand(3, 4) * 10 - 5
+    # Avoid possible division by 0 errors
+    data_tmp[data_tmp == 0] = 1.0
+    arr_data = mx.nd.array(data_tmp)
+    data = mx.symbol.Variable('data')
+    test = mx.sym.reciprocal(data)
+
+    exe_test = test.bind(default_context(), args=[arr_data])
```
Comment (piiswrong) [inline tests/python/unittest/test_operator.py]: use check_symbolic_forward and check_numerical_gradient

**Comment by piiswrong** [comment]
@apaleyes ping

**Comment by apaleyes** [comment]
Was away for few weeks. Both addressed.

Does check_symbolic_backward apply here as well?

**Pull request merged**